scripts/geninstr.ts

```diff
@@ -482,6 +576,85 @@ declare function value(id: Resource, num?: number): Value;
 declare function value(id: Frame, num?: number): Value;
 declare function value(id: Color, num?: number): Value;
 declare function value(id: Extra, num?: number): Value;
+
+type FromLink = { from: string[] };
+type ToLink = { to: string[] };
+declare function from(...links: string[]): FromLink;
+declare function to(...links: string[]): ToLink;
+
+type LinkValue = Value | AnyId | number;
+type FromLinkArg = FromLink | LinkValue | FromLink[];
+type ToLinkArg = ToLink | LinkValue | (ToLink | LinkValue)[];
+type FromToLinkArg = FromLinkArg | ToLinkArg | (FromLink | ToLink)[];
+
+// foo(v1: Value, v2: Value) -> [FromToLinkArg?, FromToLinkArg?]
+type BehaviorFromToLinkArgs<T> = Partial<{ [K in keyof T]: FromToLinkArg }> | Record<number | string, FromToLinkArg>;
+
+// https://stackoverflow.com/a/52490977
+type Tuple<T, N extends number> = N extends N ? number extends N ? T[] : _TupleOf<T, N, []> : never;
```

Comment:
Could replace these magic types by having the code generator generate the tuples instead of emitting a number that gets turned into a tuple by this type. Let me know if that's what you prefer.

Reply:
Yeah, I think that'd make the generated code easier to read.